src/stock-operations/stock-operation-types-selector/useAvailableOperationTypes.tsx

```diff
@@ -0,0 +1,53 @@
+import { useMemo } from 'react';
+import { useStockOperationTypes, useUserRoles } from '../../stock-lookups/stock-lookups.resource';
+import { useSystemSettings } from '../../core/hooks/useSystemSettings';
+
+/**
+ * Custom hook to fetch and filter available stock operation types based on user roles and system settings.
+ * @returns {{
+ *   opTypes: Array<Object>,
+ *   isLoading: boolean,
+ *   error: Error | null
+ * }} An object containing filtered operation types, loading state, and any error.
+ */
+export const useAvailableOperationTypes = () => {
+  const { stockOperationTypes: opTypes, isLoading: loadingOpTypes, error: opTypesError } = useStockOperationTypes();
+  const { userRoles } = useUserRoles();
+  const {
+    systemSettings: settings,
+    isLoading: loadingSettings,
+    error: settingsError,
+  } = useSystemSettings('stockmanagement.allowStockIssueWithoutRequisition');
+
+  /**
+   * Filter operation types based on user roles.
+   */
+  const userOpTypes = useMemo(() => {
+    if (!opTypes || !userRoles?.operationTypes) return [];
+
+    const userOpTypeIds = new Set(userRoles.operationTypes.map((role) => role.operationTypeUuid));
+    return opTypes.filter((opType) => userOpTypeIds.has(opType.uuid));
```

Comment:
can we also check for this 
https://github.com/openmrs/openmrs-module-stockmanagement/blob/b3ca338f551b0561c3beb42fc659252534b9e6f8/owa/src/features/stockOperations/Edit.tsx#L291C98-L291C158